test_optimizer.py

```
@@ -1335,3 +1336,40 @@ def test_fitness_calmar_uses_drawdown_floor():
+
+
+def test_best_trial_callback_logs_best_trial(caplog):
+    import optuna
+    from optimizer import BestTrialCallbackHandler
+
+    handler = BestTrialCallbackHandler()
+
+    class MockTrial:
+        state = optuna.trial.TrialState.COMPLETE
+        number = 5
+        values = [1.5, 2.0]
```

Ruff (0.15.9)

[warning] Mutable default value for class attribute

(RUF012)